### PR TITLE
Fix --dsjson crash

### PR DESCRIPTION
**Pull request by eisber**

### CODE REVIEW

File: vowpalwabbit/gd.cc [hunk 1]
```diff
@@ -108,17 +108,17 @@ inline void update_feature(float& update, float x, float& fw)
 
 //this deals with few nonzero features vs. all nonzero features issues.
   template<bool sqrt_rate, size_t adaptive, size_t normalized>
-  float average_update(float total_weight, float normalized_sum_norm_x, float neg_norm_power)
+  float average_update(double total_weight, double normalized_sum_norm_x, float neg_norm_power)
```
Comment (JohnLangford) [inline vowpalwabbit/gd.cc]: Why change to double?

Comment (eisber) [reply vowpalwabbit/gd.cc]: fix VS warning

**Comment by JohnLangford** [comment]
I'd rather fix this way:
https://github.com/JohnLangford/vowpal_wabbit/blob/master/vowpalwabbit/gd.cc#L491
since the downcast to float is happening just after it ceases to be an
accumulator and starts being used for general math ops.

-John

**Comment by JohnLangford** [comment]
Status on this?

**Comment by eisber** [comment]
It’s changed in the MSFT branch

-Markus

**Pull request merged**